airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/ConnectionReplicationTab.tsx

```diff
@@ -1,241 +1,171 @@
 
-interface ConnectionReplicationTabProps {
-  onAfterSaveSchema: () => void;
-  connectionId: string;
-}
-
-interface ResetWarningModalProps {
-  onClose: (withReset: boolean) => void;
-  onCancel: () => void;
-  stateType: ConnectionStateType;
-}
+export const ConnectionReplicationTab: React.FC = () => {
+  const analyticsService = useAnalyticsService();
+  const connectionService = useConnectionService();
+  const workspaceId = useCurrentWorkspaceId();
 
-const ResetWarningModal: React.FC<ResetWarningModalProps> = ({ onCancel, onClose, stateType }) => {
   const { formatMessage } = useIntl();
-  const [withReset, setWithReset] = useState(true);
-  const requireFullReset = stateType === ConnectionStateType.legacy;
-  return (
-    <>
-      <ModalBody>
-        {/* 
-        TODO: This should use proper text stylings once we have them available.
-        See https://github.com/airbytehq/airbyte/issues/14478
-      */}
-        <FormattedMessage id={requireFullReset ? "connection.streamFullResetHint" : "connection.streamResetHint"} />
-        <p>
-          <LabeledSwitch
-            checked={withReset}
-            onChange={(ev) => setWithReset(ev.target.checked)}
-            label={formatMessage({
-              id: requireFullReset ? "connection.saveWithFullReset" : "connection.saveWithReset",
-            })}
-            checkbox
-            data-testid="resetModal-reset-checkbox"
-          />
-        </p>
-      </ModalBody>
-      <ModalFooter>
-        <Button onClick={onCancel} variant="secondary" data-testid="resetModal-cancel">
-          <FormattedMessage id="form.cancel" />
-        </Button>
-        <Button onClick={() => onClose(withReset)} data-testid="resetModal-save">
-          <FormattedMessage id="connection.save" />
-        </Button>
-      </ModalFooter>
-    </>
-  );
-};
+  const { openModal } = useModalService();
 
-export const ConnectionReplicationTab: React.FC<ConnectionReplicationTabProps> = ({
-  onAfterSaveSchema,
-  connectionId,
-}) => {
-  const { formatMessage } = useIntl();
-  const { openModal, closeModal } = useModalService();
-  const { openConfirmationModal, closeConfirmationModal } = useConfirmationModalService();
-  const [activeUpdatingSchemaMode, setActiveUpdatingSchemaMode] = useState(false);
   const [saved, setSaved] = useState(false);
-  const connectionService = useConnectionService();
+
+  const formId = useUniqueFormId();
+  const { connection, schemaRefreshing, schemaHasBeenRefreshed, updateConnection, setSchemaHasBeenRefreshed } =
+    useConnectionEditService();
+  const { initialValues, mode, schemaError, getErrorMessage, setSubmitError } = useConnectionFormService();
+
   useTrackPage(PageTrackingCodes.CONNECTIONS_ITEM_REPLICATION);
 
-  const { mutateAsync: updateConnection } = useUpdateConnection();
+  const saveConnection = useCallback(
+    async (values: ValuesProps, skipReset: boolean, catalogHasChanged: boolean) => {
```

Comment:
Can we group the 2nd and 3rd parameter into an options object please. Boolean parameters are very hard to read by itself, so we'll end up with code like `saveConnection(values, true, false)` that is not understanable from the place it's read. Thus was the original reason this was an object for `skipReset`.